conduit/pipeline/pipeline.go

```diff
@@ -57,6 +57,14 @@ type Config struct {
 	Processors []NameConfigPair `yaml:"processors"`
 	Exporter   NameConfigPair   `yaml:"exporter"`
 	Metrics    Metrics          `yaml:"metrics"`
+	// RetryCount is the number of retries to perform for an error in the pipeline
+	RetryCount uint64 `yaml:"retry-count"`
```

Comment:
nit: does retry count need to be so big? should we set a default max?

Reply:
The default is 10 and it is set in `MakePipelineConfig()`.  I chose uint64 just out of habit.  I'm fine with whatever integer type as long as it is unsigned.